xlators/mgmt/glusterd/src/glusterd-locks.c

```diff
@@ -584,6 +585,8 @@ glusterd_mgmt_v3_lock(const char *name, uuid_t uuid, uint32_t *op_errno,
     if (ret) {
         gf_msg(this->name, GF_LOG_ERROR, 0, GD_MSG_DICT_SET_FAILED,
                "Unable to set timer in mgmt_v3 lock");
+        keylen = strlen(key_dup);
+        dict_deln(priv->mgmt_v3_lock, key_dup, keylen);
```

Comment:
Shouldn't you also free  'lock_obj' ?

nit: the whole keylen and such and using dict_deln()  - you could just use dict_del()

Reply:
@mykaul I have made modifications based on the suggestions. Please review it again.
Thanks!